syntax = "proto3";

package receipt_verifier;

service ReceiptVerifierService {
  rpc VerifyReceiptStream (stream BytesChunk) returns (GrpcVerifyResponse);
}


message BytesChunk {
  bytes data = 1;
}

message GrpcVerifyResponse {
  bool valid = 1;
  string message = 2;
  optional double journal_value = 3;
}